org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/handlers/CodeLensHandler.java

```diff
@@ -80,9 +82,12 @@ public CodeLens resolve(CodeLens lens, IProgressMonitor monitor) {
 			command = JAVA_SHOW_IMPLEMENTATIONS_COMMAND;
 		}
 		try {
-			ICompilationUnit unit = JDTUtils.resolveCompilationUnit(uri);
-			if (unit != null) {
-				IJavaElement element = JDTUtils.findElementAtSelection(unit, ((Double) position.get("line")).intValue(), ((Double) position.get("character")).intValue(), this.preferenceManager, monitor);
+			ITypeRoot typeRoot = JDTUtils.resolveCompilationUnit(uri);
+			if (typeRoot == null) {
+				typeRoot = JDTUtils.resolveClassFile(uri);
+			}
```

Comment:
maybe we should add a `JDTUtils.resolveTypeRoot(uri)` method instead, check were else it would apply

Reply:
There is already such a method - https://github.com/eclipse/eclipse.jdt.ls/blob/master/org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/JDTUtils.java#L271